core/silkworm/execution/execution.cpp

```diff
@@ -22,11 +22,9 @@
 
 namespace silkworm {
 
-std::pair<std::vector<Receipt>, ValidationResult> execute_block(const Block& block, StateBuffer& buffer,
-                                                                const ChainConfig& config,
-                                                                AnalysisCache* analysis_cache,
-                                                                ExecutionStatePool* state_pool,
-                                                                evmc_vm* exo_evm) noexcept {
+ValidationResult execute_block(const Block& block, StateBuffer& buffer, const ChainConfig& config,
+                               std::vector<Receipt>& receipts, AnalysisCache* analysis_cache,
+                               ExecutionStatePool* state_pool, evmc_vm* exo_evm) noexcept {
```

Comment:
Just a question: why not make execute_block a std::pair<std::vector<receipt>, ValidationResult> ?
If I read it correctly the reference to receipts is passed through all the stack where `ExecutionProcessor::execute_block` eventually clears it and reserve. I.e. basically is ExecutionProcessor::execute_block which creates the instance of vector.

Reply:
> Just a question: why not make execute_block a std::pair<std::vector, ValidationResult> ?
> If I read it correctly the reference to receipts is passed through all the stack where `ExecutionProcessor::execute_block` eventually clears it and reserve. I.e. basically is ExecutionProcessor::execute_block which creates the instance of vector.

Well, yes, `ExecutionProcessor::execute_block` clears the receipts, but it doesn't call `shrink_to_fit`, so it avoids deallocating memory and benefits from its subsequent reuse.